publish-wrapper

```diff
@@ -0,0 +1,204 @@
+        ret = task.run(opts)
+    except RuntimeError as ex:
+        ret = str(ex)
+
+    if ret:
+        sys.stderr.write("log-wrapper: {0}\n".format(ret))
+        return 1
+    return 0
+
+
+class PullTask():
+    def __init__(self, base_project):
+        self.sink = None
+        self.github_status_data = None
+        self.api = github.GitHub(repo=base_project)
+
+    def start_publishing(self, opts):
+        # build a unique file name for this test run
+        id_context = opts.github_context.replace("@", "-").replace("#", "-")
+        identifier = "-".join([
+            opts.test_name.replace("/", "-"),
+            opts.revision[0:8],
+            id_context.replace("/", "-")
+        ])
+
+        description = "{0} [{1}]".format(github.TESTING, HOSTNAME)
+
+        self.github_status_data = {
+            "state": "pending",
+            "context": opts.github_context,
+            "description": description,
+            "target_url": ":link"
+        }
+
+        status = {
+            "github": {
+                "token": self.api.token,
+                "requests": [
+                    # Set status to pending
+                    {
+                        "method": "POST",
+                        "resource": self.api.qualify("commits/" + opts.revision + "/statuses"),
+                        "data": self.github_status_data
+                    }
+                ],
+                "watches": [{
+                    "resource": self.api.qualify("commits/" + opts.revision + "/status?per_page=100"),
+                    "result": {
+                        "statuses": [
+                            {
+                                "context": opts.github_context,
+                                "description": description,
+                                "target_url": ":link"
+                            }
+                        ]
+                    }
+                }]
+            },
+            "revision": opts.revision,
+            "onaborted": {
+                "github": {
+                    "token": self.api.token,
+                    "requests": [
+                        # Set status to error
+                        {
+                            "method": "POST",
+                            "resource": self.api.qualify("statuses/" + opts.revision),
+                            "data": {
+                                "state": "error",
+                                "context": opts.github_context,
+                                "description": "Aborted without status",
+                                "target_url": ":link"
+                            }
+                        }
+                    ]
+                },
+            }
+        }
+
+        status["link"] = "log.html"
+        status["extras"] = ["https://raw.githubusercontent.com/cockpit-project/bots/master/task/log.html"]
+
+        # For other scripts to use
+        os.environ["TEST_DESCRIPTION"] = description
```

Comment:
I can't find any place which uses this env var, so this is another potential cleanup target.